src/EFCore.SqlServer/Metadata/Conventions/SqlServerValueGenerationConvention.cs

```diff
@@ -53,6 +53,26 @@ public class SqlServerValueGenerationConvention : RelationalValueGenerationConve
             base.ProcessPropertyAnnotationChanged(propertyBuilder, name, annotation, oldAnnotation, context);
         }
 
+        /// <inheritdoc />
+        public override void ProcessEntityTypeAnnotationChanged(
+            IConventionEntityTypeBuilder entityTypeBuilder,
+            string name,
+            IConventionAnnotation? annotation,
+            IConventionAnnotation? oldAnnotation,
+            IConventionContext<IConventionAnnotation> context)
+        {
+            if ((name == SqlServerAnnotationNames.TemporalPeriodStartPropertyName
+                || name == SqlServerAnnotationNames.TemporalPeriodEndPropertyName)
+                && annotation != null
+                && annotation.Value is string propertyName
+                && entityTypeBuilder.Metadata.FindProperty(propertyName) is IConventionProperty property)
+            {
+                property.SetValueGenerated(ValueGenerated.OnAddOrUpdate);
```

Comment:
This won't reset the `ValueGenerated` configuration to what it was before if the table is configured to not be temporal (I know this is extremely corner-case, but we should be consistent when possible).

Instead do
```suggestion
                property.Builder.ValueGenerated(GetValueGenerated(property));
```

and modify `GetValueGenerated`

Reply:
when we convert back from temporal to normal table we should probably remove the period properties/columns - will add the test for this (if we don't have one already). But we also need to consider the scenario where user property is mapped to the period column